File: fibonacci/julia/08_linear/fibonacci.jl
function fibonacci(n)
    v0 = 0
    v1 = 1
    if n < 0
        for _ ∈ -1:-1:n
            (v0, v1) = (v1 - v0, v0)
        end
        v0
    elseif n == 0
        v0
    elseif n == 1
        v1
    else
        for _ ∈ 2:n
            (v1, v0) = (v0 + v1, v1)
        end
        v1
    end
end

File: fibonacci/python/08_linear/fibonacci.py
def fibonacci(n):
    v0 = 0
    v1 = 1
    if n < 0:
        for _ in range(-1, n - 1, -1):
            v0, v1 = (v1 - v0, v0)
        return v0
    if n == 0:
        return v0
    if n == 1:
        return v1
    for _ in range(2, n + 1):
        v1, v0 = (v0 + v1, v1)
    return v1
